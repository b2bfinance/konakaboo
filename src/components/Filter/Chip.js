import styled, { css } from 'styled-components';
import Chip from '@material-ui/core/Chip';

export { Chip as BaseChip };

export const ChipHolder = styled.div`
  margin-bottom: 1.5rem;
  margin-right: 0.5rem;
  position: relative;
`;

export default styled(Chip)`
  ${props =>
    props.selection.length > 0 &&
    css`
      background-color: ${props.theme.filterChipChosenBackground} !important;
      color: ${props.theme.filterChipChosenColor} !important;
    `};
`;
