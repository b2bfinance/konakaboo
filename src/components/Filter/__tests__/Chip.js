import React from 'react';
import { render } from 'enzyme';
import Chip, { ChipHolder } from '../Chip';

describe('ChipHolder', () => {
  test('renders correctly', () => {
    expect(render(<ChipHolder />)).toMatchSnapshot();
  });
});

describe('Chip', () => {
  test('renders correctly with no selection', () => {
    expect(render(<Chip selection={[]} />)).toMatchSnapshot();
  });

  test('renders correctly with selection', () => {
    expect(render(<Chip selection={[0, 1]} />)).toMatchSnapshot();
  });
});
