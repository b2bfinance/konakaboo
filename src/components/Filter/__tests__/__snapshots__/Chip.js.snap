// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders correctly with no selection 1`] = `
<div
  class="MuiChip-root-1 "
  role="button"
  selection=""
  tabindex="-1"
>
  <span
    class="MuiChip-label-20"
  />
</div>
`;

exports[`Chip renders correctly with selection 1`] = `
.c0 {
  background-color: !important;
  color: !important;
}

<div
  class="MuiChip-root-1 c0"
  role="button"
  selection="0,1"
  tabindex="-1"
>
  <span
    class="MuiChip-label-20"
  />
</div>
`;

exports[`ChipHolder renders correctly 1`] = `
.c0 {
  margin-bottom: 1.5rem;
  margin-right: 0.5rem;
  position: relative;
}

<span
  class="c0"
/>
`;
